controllers/machinehealthcheck_controller.go

```diff
@@ -250,7 +250,8 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 	// mark for remediation
 	errList := []error{}
 	for _, t := range unhealthy {
-		logger.V(3).Info("Target meets unhealthy criteria, triggers remediation", "target", t.string())
+		condition := conditions.Get(t.Machine, clusterv1.MachineHealthCheckSuccededCondition)
```

Comment:
Is this guaranteed to be non-nil? I'd assume so, just double checking

Reply:
Everything in `unhealthy` should be getting its condition set in `needsRemediation()`